### ISSUE TITLE
osx compile error when link to OgreOverlay

### ISSUE THREAD

**Issue by dandingol03** [OGRECave/ogre#2144]
#### System Information
- Ogre Version: :1.12.12 
- Operating System / Platform:: macOS Big Sur 11.4

#### Detailled description

 After i configure and generate xcode project, i run the `ALL_BUILD` target. there is some compiled error which under `OgreOverlay` package. it seems like that the code can't solve the freetype dependency. Looking forward to your reply

#### Ogre.log

```
Undefined symbols for architecture x86_64:
"_inflate", referenced from:
_FT_Gzip_Uncompress in libfreetype.a(ftgzip.o)
_ft_gzip_file_fill_output in libfreetype.a(ftgzip.o)
"_inflateEnd", referenced from:
_FT_Stream_OpenGzip in libfreetype.a(ftgzip.o)
_ft_gzip_stream_close in libfreetype.a(ftgzip.o)
_FT_Gzip_Uncompress in libfreetype.a(ftgzip.o)
"inflateInit2", referenced from:
_FT_Stream_OpenGzip in libfreetype.a(ftgzip.o)
_FT_Gzip_Uncompress in libfreetype.a(ftgzip.o)
"_inflateReset", referenced from:
_FT_Stream_OpenGzip in libfreetype.a(ftgzip.o)
_ft_gzip_file_io in libfreetype.a(ftgzip.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

**Comment by paroj** [comment]
you can try setting OGRE_BUILD_DEPENDENCIES=FALSE in cmake and install the dependencies manually via brew.
Alternatively, try the master branch where the freetype dependency is optional.

**Comment by paroj** [comment]
fixed in #2163
